Content.Shared/Cloning/CloningPodComponent.cs

```diff
@@ -0,0 +1,137 @@
+using Content.Shared.Construction.Prototypes;
+using Content.Shared.Materials;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Cloning;
+[RegisterComponent]
+public sealed partial class CloningPodComponent : Component
+{
+    public const string PodPort = "CloningPodReceiver";
+
+    [ViewVariables]
+    public ContainerSlot BodyContainer = default!;
+
+    /// <summary>
+    /// How long the cloning has been going on for.
+    /// </summary>
+    [ViewVariables]
+    public float CloningProgress = 0;
+
+    [ViewVariables]
+    public int UsedBiomass = 70;
+
+    [ViewVariables]
+    public bool FailedClone = false;
+
+    /// <summary>
+    /// The material that is used to clone entities.
+    /// </summary>
+    [DataField("requiredMaterial", customTypeSerializer: typeof(PrototypeIdSerializer<MaterialPrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    public string RequiredMaterial = "Biomass";
```

Comment:
can do what I said above for datafields as well so you don't type that long ass custom type serializer